Allow a list of algorithms for decode

lautis opened the pull request:
Hard-coding only one algorithms makes rolling out changes harder. Both ends need to be deployed simultaneously or retry logic needs to be implemented in app.

Allowing the algorithms to be a list containing potentially multiple entries enables flexibility when changing algorithms. The default value is left unchanged and giving `:algorithm` option instead of  `:algorithms` still works.

excpt commented:
@lautis Thanks for the PR. I left you a comment to improve the code.

lautis commented:
@excpt should that be visible somewhere?

excpt commented:
@lautis Now it should be visible. I need to press the "submit review"-button in order to make my review visible.

lautis commented on spec/jwt_spec.rb:
```diff
@@ -231,18 +231,30 @@
 
         expect do
-          JWT.decode token, data[:secret], true, algorithm: 'HS384'
+          JWT.decode token, data[:secret], true, algorithms: 'HS384'
```
That's a typo, should be left as `algorithm` there.

excpt replied:
Simply push a fix for that and I'll merge it.

lautis commented:
I added one more change: removed the combination of option because it could lead into the default algorithms array being used in addition to the value given with algorithm option.

excpt commented:
Thanks again for your contribution. 🎉

lautis commented:
Thank you for maintaining JWT! 🙏
